### DISCUSSION TITLE
Consistency levels in Stargate

### DISCUSSION THREAD

**Discussion by jwd24** [stargate/stargate#1886] [Help] [answered]
I'm struggling with understanding consistency levels in Stargate.
We are using 3 Cassandra nodes in a single DC.
We are using the Document API to create our keyspaces/namespaces and tables and have a replication factor of 3.
Q1.  By default, what consistency level will be used for these keyspaces/namespaces?
Q2.  I've read that consistency levels can be set on individual reads/writes, but I don't see anywhere in the Document API that says how to do that.  Can someone please explain or show an example?

Thanks,
John

**Comment by EricBorczuk** [comment, marked as answer]
Hi John! Thanks for reaching out.

The consistency level used by the entire Documents API is LOCAL_QUORUM. Keeping in mind that the Documents API is a layer on top of Cassandra, and that it does a bunch of its own operating on rows in order to store document rows and keep them in a proper state, it is actually opinionated about the consistency level, and it is NOT configurable.

The other API's in Stargate (graphql, CQL, plain REST) all allow configuring the consistency level on each request and default to a level of LOCAL_QUORUM when applicable.

Hope this helps!

**Reply by jwd24** [reply]
Hi Eric,

Thanks for the quick reply.  I'm glad to hear that the consistency is LOCAL_QUORUM by default, but then I'm confused by some test results I'm seeing.  We have a JMeter test doing Adds, Gets and Deletes with the database (in a Kubernetes environment).
When we do a drain (which takes down one of the 3 Cassandra nodes), we get a few failures, but with LOCAL_QUORUM, I would have expected this to work without failure.  Maybe there is something else wrong in our environment.

**Reply by EricBorczuk** [reply]
If you have more detailed information about those failures and it points to an issue with the API, feel free to open up an issue and we can dive into it deeper!

**Reply by jwd24** [reply]
Awesome!  Thanks Eric.  Currently the systems we are seeing these errors on are being used by another group for the next week or so, but once we get the systems back, if we are still having this problem, I'll provide more details on this thread.

**Comment by ivansenic** [comment]
@jwd24 Be aware that in Docs API V2 you will be able to set consistency using the configuration. See https://github.com/stargate/stargate/blob/v2.0.0/sgv2-docsapi/CONFIGURATION.md#queries-configuration for more details.
